inc/saidot1brfdb.h

```diff
@@ -0,0 +1,130 @@
+/*
+* Copyright (c) 2015 Dell Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may
+* not use this file except in compliance with the License. You may obtain
+* a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+* THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+* LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+* FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+*
+* See the Apache Version 2.0 License for specific language governing
+* permissions and limitations under the License.
+*
+* @file saidot1brfdb.h
+*
+* @brief This module defines SAI API for IEEE 802.1BR ECID based forwarding
+         functionality. This is applicable only to Port Extenders (PE).
+*************************************************************************************/
+
+#if !defined (__SAIDOT1BRFDB_H)
+#define __SAIDOT1BRFDB_H
+
+#include "saitypes.h"
+#include "saistatus.h"
+
+/** \defgroup SAIDOT1BRFDBENTRY SAI - 802.1BR ECID based forwarding specific public APIs and datastructures in PE.
+ *
+ *  \{
+ */
+
+/**
+ * @brief SAI attributes for SAI_OBJECT_TYPE_DOT1BR_FDB_ENTRY 
+ */
+typedef enum _sai_dot1br_fdb_entry_attr_t
```

Comment:
Dell - Internal Use - Confidential
Hi Kamil Cudnik,

This FDB table is the forwarding table in Port Extender (PE). There are quite a number differences between the native L2 Fdb table and this one and hence it is not combined with the native L2 Fdb table.


1.      There is no learning and all entries are static

2.      Flush table requests are not applicable

3.      This table contains both unicast and multicast entries. In case of multicast entries, it will contain a list of ports.


This pull request is an old pull request. With respect to the SAI pipeline model and Bridge port proposals, the dot1br proposal required quite a number of changes. These changes were reflected in the proposal document. The changes in the dot1br header files cannot be incorporated until the bridge port header file changes come in. So waiting for those changes to update the dot1br pull request.

So the I have to come up with an updated dot1br proposal.

I did not want to delete the pull request, in order to preserve the comments provided on the pull request.

Regards,
Ravi

From: Kamil Cudnik [mailto:notifications@github.com]
Sent: Wednesday, August 24, 2016 9:45 AM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Sivasankar, Ravikumar <Ravikumar_Sivasankar@Dell.com>; Author <author@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] IEEE 802.1BR SAI proposal. (#73)


In inc/saidot1brfdb.h<https://github.com/opencomputeproject/SAI/pull/73#discussion_r75992553>:

> +

> +#if !defined (__SAIDOT1BRFDB_H)

> +#define __SAIDOT1BRFDB_H

> +

> +#include "saitypes.h"

> +#include "saistatus.h"

> +

> +/** \defgroup SAIDOT1BRFDBENTRY SAI - 802.1BR ECID based forwarding specific public APIs and datastructures in PE.

> + *

> + *  \{

> + */

> +

> +/**

> + * @brief SAI attributes for SAI_OBJECT_TYPE_DOT1BR_FDB_ENTRY

> + */

> +typedef enum _sai_dot1br_fdb_entry_attr_t

there is plan to convert fdb_entry_t to actual sai_object_id_t and mac and vlan could mandatory attributes, Type attribute could be introduced as mandatory attribute to determine wheter its fdb entry or fdb_dot1br entry and based on type only specific attributes would apply to specific fdb entry type.
IMO we dont need more unions, they make things more complicated and next to union we still need a type to know which union member is used. it would be better to use attributes instead

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/73/files/4302a4c03dbc322a2d63ce4cc56895a3d50bf984#r75992553>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AOoHu6urOGlehx-UM-3NASNpjHFI--Qoks5qi8UrgaJpZM4HpSGE>.